cmake/FindJSONC.cmake

```diff
@@ -0,0 +1,22 @@
+# - Try to find JSON-C
+# Once done this will define
+#
+#  JSONC_FOUND - JSON-C was found
+#  JSONC_INCLUDE_DIRS - JSON-C include directories
+#  JSONC_LIBRARIES - libraries needed for linking
+
+find_package(PkgConfig REQUIRED)
```

Comment:
this will abort configuration on a system without pkg-config

Reply:
Some of the dependencies already require pkg-config, but it is true that they are not built by default, or only on Linux. So I will change that to not require it. Also the `pkg_check_modules` result was not used by mistake, so I will add the missing `HINTS`...